src/main/java/de/doubleslash/keeptime/Main.java

```diff
@@ -233,6 +245,51 @@ private void initialisePopupUI(final Stage primaryStage) throws IOException {
 
    private void initialiseUI(final Stage primaryStage) throws IOException {
       LOG.debug("Initialising main UI.");
+
+      final ScreenPosHelper poshelper = new ScreenPosHelper(model.screenSettings.screenHash.get(),
+            model.screenSettings.xProportion.get(), model.screenSettings.yProportion.get());
+      poshelper.resetPositionIfInvalid();
+
+      // set stage to saved Position
+      if (model.screenSettings.saveWindowPosition.get()) {
+         primaryStage.setX(poshelper.getAbsoluteX());
+         primaryStage.setY(poshelper.getAbsoluteY());
+      }
+
+      // add listeners to record Windowpositionchange
+
+      final ChangeListener<Number> posChange = new ChangeListener<Number>() {
+
+         @Override
+         public void changed(final ObservableValue<? extends Number> observable, final Number oldValue,
+               final Number newValue) {
+
+            // dont save if option disabled
+            if (!model.screenSettings.saveWindowPosition.get()) {
+               return;
+            }
+
+            if (observable.equals(primaryStage.xProperty())) {
+               poshelper.setAbsoluteX(newValue.doubleValue());
+            } else {
+               poshelper.setAbsoluteY(newValue.doubleValue());
+            }
+
+            final Settings newSettings = new Settings(model.hoverBackgroundColor.get(), model.hoverFontColor.get(),
+                  model.defaultBackgroundColor.get(), model.defaultFontColor.get(), model.taskBarColor.get(),
+                  model.useHotkey.get(), model.displayProjectsRight.get(), model.hideProjectsOnMouseExit.get(),
+                  poshelper.getProportionalX(), poshelper.getProportionalY(), poshelper.getScreenHash(),
+                  model.screenSettings.saveWindowPosition.get(), model.remindIfNotesAreEmpty.get());
+
+            controller.updateSettings(newSettings);
```

Comment:
I'm not too confident that saving this to the db on each move is good. If you drag and drop the app there are loads of events (also x and y are separate events).
Guess we should just save it once when shutting down, and later also when autoSave is merged..

Reply:
implemented to just save on shutdown